airflow/utils/state.py

```diff
@@ -120,3 +120,10 @@ def unfinished(cls):
             cls.UP_FOR_RETRY,
             cls.UP_FOR_RESCHEDULE
         ]
+
+    @classmethod
+    def unsuccessful(cls):
+        """
+        A list of states indicating that a task completed unsuccessfully.
+        """
+        return [cls.FAILED, cls.UP_FOR_RETRY]
```

Comment:
Please make this a set
```suggestion
        return {cls.FAILED, cls.UP_FOR_RETRY}
```

Reply:
All other class method in this class is using an array, I'd prefer to keep it consistent